ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/events/filesystem/FSTableEvent.java

```diff
@@ -186,6 +188,17 @@ private AddPartitionDesc partitionDesc(Path fromPath,
             Warehouse.makePartName(tblDesc.getPartCols(), partition.getValues())).toString());
       }
       partsDesc.setReplicationSpec(replicationSpec());
+
+      // Right now, we do not have a way of associating a writeId with statistics for a table
+      // converted to a transactional table if it was non-transactional on the source. So, do not
+      // update statistics for converted tables even if available on the source.
+      if (partition.isSetColStats() && !replicationSpec().isMigratingToTxnTable()) {
+        ColumnStatistics colStats = partition.getColStats();
+        ColumnStatisticsDesc colStatsDesc = new ColumnStatisticsDesc(colStats.getStatsDesc());
+        colStatsDesc.setDbName(tblDesc.getTableName());
```

Comment:
Good catch. Fixed the bug. Since replication doesn't change the name, and we set the correct dbname in just the next line the bug was not caught. Thanks. Corrected.